app/connector/debezium/src/main/java/io/syndesis/connector/debezium/DebeziumMetaDataRetrieval.java

```diff
@@ -0,0 +1,168 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.connector.debezium;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.StringJoiner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.syndesis.common.model.DataShape;
+import io.syndesis.common.model.DataShapeKinds;
+import io.syndesis.connector.kafka.KafkaMetaDataRetrieval;
+import io.syndesis.connector.support.util.ConnectorOptions;
+import io.syndesis.connector.support.verifier.api.PropertyPair;
+import io.syndesis.connector.support.verifier.api.SyndesisMetadata;
+import org.apache.camel.CamelContext;
+import org.apache.camel.component.extension.MetaDataExtension;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@SuppressWarnings("PMD")
+public class DebeziumMetaDataRetrieval extends KafkaMetaDataRetrieval {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DebeziumMetaDataRetrieval.class);
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+    private static final DataShape ANY = new DataShape.Builder().kind(DataShapeKinds.ANY).build();
+
+    @Override
+    protected SyndesisMetadata adapt(CamelContext context, String componentId, String actionId, Map<String, Object> properties, MetaDataExtension.MetaData metadata) {
+        // Retrieve the list of topics
+        @SuppressWarnings("unchecked")
+        Set<String> topicsNames = (Set<String>) metadata.getPayload();
+        List<PropertyPair> topicsResult = new ArrayList<>();
+        topicsNames.stream().forEach(
+            t -> topicsResult.add(new PropertyPair(t, t))
+        );
+        // Retrieve dynamically from list of topics provided by Kafka
+        Map<String, List<PropertyPair>> dynamicProperties = new HashMap<>();
+        dynamicProperties.put("topic", topicsResult);
+        dynamicProperties.put("schemaChange", topicsResult);
+
+        final String brokers = ConnectorOptions.extractOption(properties, "brokers");
+        final String topicSelected = ConnectorOptions.extractOption(properties, "topic");
+        final String schemaChangeSelected = ConnectorOptions.extractOption(properties, "schemaChange");
+
+        DataShape outputDataShape = topicSelected != null ? getDatashape(brokers, topicSelected, schemaChangeSelected) : ANY;
+        return new SyndesisMetadata(
+            dynamicProperties,
+            null,
+            outputDataShape
+        );
+    }
+
+    @SuppressWarnings("PMD")
```

Comment:
Please don't blanket disable PMD, suppress only individual issues when there's no way to fix them